packages/cli/src/verifyContract.ts

```diff
@@ -0,0 +1,57 @@
+import { forge } from "@latticexyz/common/foundry";
+import { Address, ByteArray, Hex, getCreate2Address } from "viem";
+import { salt } from "./deploy/common";
+
+type VerificationOptions = {
+  address: Hex;
+  name: string;
+  rpc: string;
+  verifier?: string;
+  verifierUrl?: string;
+};
+
+type ForgeOptions = { profile?: string; silent?: boolean; env?: NodeJS.ProcessEnv; cwd?: string };
+
+type Create2Options = {
+  bytecode: ByteArray | Hex;
+  from: Address;
+  salt: ByteArray | Hex;
+};
+
+export async function verifyContract(
+  { address, name, rpc, verifier, verifierUrl }: VerificationOptions,
+  options?: ForgeOptions,
+) {
+  const args = ["verify-contract", address, name, "--rpc-url", rpc];
+
+  if (verifier) {
+    args.push("--verifier", verifier);
+  }
+  if (verifierUrl) {
+    args.push("--verifier-url", verifierUrl);
+  }
+  await forge(args, options);
+}
+
+export async function verifyContractCreate2(
+  { name, rpc, verifier, verifierUrl, from, bytecode, salt }: Omit<VerificationOptions, "address"> & Create2Options,
```

Comment:
What if we just make the options this:

```ts
type VerifyContractOptions = {
  foundryProfile?: string;
  verifier?: string;
  verifierUrl?: string;
} & ({ address: Hex; } | Create2Options);

```

Then we would only need one `verifyContract` function instead of a different path for `verifyContractCreate2`.

Reply:
Oh that's wayyy better. 

my brain lives in Solidity-land where you create overloads for each set of arguments